### PR TITLE
updating the approval requirement

### PR DESCRIPTION
**Pull request by dhrubo-os**
### Description
[updating the approval requirement so that it can trigger always for opensearch-trigger-bot[bot]]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/require-approval.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
             # Handle Pull Request events
 
             # Check if the PR is triggered by opensearch-trigger-bot
-            if [[ "$github_event" == "pull_request" ]] && [[ "$author" == "opensearch-trigger-bot[bot]" ]]; then
+            if [[ "$author" == "opensearch-trigger-bot[bot]" ]]; then
```
Comment (ylwu-amzn) [inline .github/workflows/require-approval.yml:32]: Just curious, this author `opensearch-trigger-bot[bot]` should be some Github unique user ID , right ? Is it possible someone else bypass with same name?

Comment (dhrubo-os) [reply .github/workflows/require-approval.yml:32]: the opensearch-trigger-bot[bot] is a unique bot account managed by GitHub, and its identity is authenticated by GitHub's infrastructure. Regular users cannot impersonate it or spoof its actions by just creating an account with a similar name.

**Review by ylwu-amzn — APPROVED** [review]

**Review by yuye-aws — APPROVED** [review]
Looks good to me

**Comment by yuye-aws** [comment]
Flakey test here: https://github.com/opensearch-project/ml-commons/actions/runs/11470521177/job/31921759380?pr=3148

**Review by mingshl — APPROVED** [review]

**Pull request merged**